### DISCUSSION TITLE
Is it intentional to show all css and js dependencies regardless which component is used in the template / webpage?

### DISCUSSION THREAD

**Discussion by simkimsia** [django-components/django-components#161] [Q&A] [answered]
I realized ALL the css and js dependencies will be displayed at 

```html
<head>
    <title>Demo Tagify Basic using Django Components</title>
    <!-- the component_css_dependencies will drag out ALL the css files -->
    {% component_css_dependencies %}
</head>
<body>
    {% component "tagify_basic" date="2015-06-19" %}
    <!-- the component_js_dependencies will drag out ALL the js files -->
    {% component_js_dependencies %}
    <script>
      window.addEventListener('load', function () {
        {% include "tagify/00_basic/tagify.js" %}
      });
    </script>
</body>
```

Am I right that this is intentional?

**Comment by EmilStenstrom** [comment, marked as answer]
Yes, this is intentional, and will mean everything is cached on the next load. There is code to conditionally load component media, but it's currently not documented.

**Reply by simkimsia** [reply]
Do you mind documenting the conditional load? No hurry of course.

**Comment by EmilStenstrom** [comment]
Yeah, there's a issue for that here: https://github.com/EmilStenstrom/django-components/issues/71

**Comment by simkimsia** [comment]
> Yeah, there's a issue for that here: https://github.com/EmilStenstrom/django-components/issues/71

Thanks subscribed
